src/es/managers/CalloutPlacementManager.js

```diff
@@ -0,0 +1,227 @@
+    calculateCalloutPosition(targetElBox, calloutElBox, isRtl, arrowWidth) {
+      let top = targetElBox.top;
+      let left = targetElBox.right + arrowWidth;
+      return { top, left };
+    },
+    setArrowPosition: setArrowPositionHorizontal
+  }
+};
+
+
+/**
+  * If step is right-to-left enabled, flip the placement and xOffset.
+  * Will adjust placement only once and will set _isFlippedForRtl option to keep track of this
+  * If placement is set on a global or tour level and callout does not have this config
+  * tour\global config will stay intact and new setting will be added into the callout's config
+  * @private
+  */
+function adjustPlacementForRtl(callout, placementStrategy) {
+  let isRtl = callout.config.get('isRtl');
+  let isFlippedForRtl = callout.config.get('_isFlippedForRtl');
+
+  if (isRtl && !isFlippedForRtl) {
+    let calloutXOffset = callout.config.get('xOffset');
+    let rtlPlacement = placementStrategy.rtlPlacement;
+
+    //flip xOffset
+    if (calloutXOffset) {
+      callout.config.set('xOffset', -1 * Utils.getPixelValue(calloutXOffset));
+    }
+    //flip placement for right and left placements only
+    if (rtlPlacement) {
+      callout.config.set('placement', rtlPlacement);
+      placementStrategy = placementStrategies[rtlPlacement];
+    }
+  }
+  return placementStrategy;
+}
+
+/**
+  * Adds correct placement class to the callout's arrow element if it exists.
+  * Arrow class will be determined based on Callout's 'placement' option.
+  * @private
+  */
+function positionArrow(callout, placementStrategy) {
+  let arrowEl = callout.el.querySelector('.hopscotch-arrow');
+  if (!arrowEl) {
+    return;
+  }
+
+  //Remove any stale position classes
+  Utils.removeClass(arrowEl, 'down up right left');
+  //Have arrow point in the direction of the target
+  Utils.addClass(arrowEl, placementStrategy.arrowPlacement);
+
+  //Position arrow correctly relative to the callout
+  let arrowOffset = callout.config.get('arrowOffset');
+  let horizontalProp = callout.config.get('isRtl') ? 'right' : 'left';
+  placementStrategy.setArrowPosition(arrowEl, callout.el, horizontalProp, arrowOffset);
+}
+
+
+/**
+ * 
+ * 
+ */
+function positionCallout(callout, placementStrategy) {
+  let targetEl = Utils.getTargetEl(callout.config.get('target'));
+  if (!targetEl) {
+    return;
+  }
+
+  let isFixedEl = callout.config.get('fixedElement');
+  let targetElBox = targetEl.getBoundingClientRect();
+  let calloutElBox = { width: callout.el.offsetWidth, height: callout.el.offsetHeight };
+  let calloutPosition = placementStrategy.calculateCalloutPosition(
+    targetElBox,
+    calloutElBox,
+    callout.config.get('isRtl'),
+    callout.config.get('arrowWidth')
+    );
+  
+  //Adjust position if xOffset and yOffset are specified
+  //horizontal offset
+  let xOffset = callout.config.get('xOffset');
+  if (xOffset === 'center') {
+    calloutPosition.left = (targetElBox.left + targetEl.offsetWidth / 2) - (calloutElBox.width / 2);
```

Comment:
Good catch! I'll address this in the next PR, since I need to add unit tests for xOffset and yOffset and rework css for the arrow (right now it's size is not correctly set).  It's will be pretty extensive change on it's own and I don't want this PR to keep growing.